src/jsdoc.ts

```diff
@@ -150,15 +150,24 @@ const JSDOC_TAGS_INPUT_BLACKLIST = new Set([
   'record',   'static',     'template',   'this',        'type',      'typedef',
 ]);
 
+/** What to do with types found in JSDoc @tags. */
+enum TagWithTypePolicy {
+  FORBIDDEN,
+  OPTIONAL,
+  REQUIRED,
+}
+
 /**
  * A list of JSDoc @tags that might include a {type} after them. Only banned when a type is passed.
  * Note that this does not include tags that carry a non-type system type, e.g. \@suppress.
  */
-const JSDOC_TAGS_WITH_TYPES = new Set([
-  'const',
-  'export',
-  'param',
-  'return',
+const JSDOC_TAGS_WITH_TYPES = new Map([
+  ['const', TagWithTypePolicy.FORBIDDEN],
+  ['define', TagWithTypePolicy.OPTIONAL],  // NOTE: this could be made forbidden later.
+  ['export', TagWithTypePolicy.FORBIDDEN],
+  ['param', TagWithTypePolicy.FORBIDDEN],
+  ['return', TagWithTypePolicy.FORBIDDEN],
+  ['suppress', TagWithTypePolicy.REQUIRED],
```

Comment:
Add a comment that this is required, because the thing in the type position is not a "type"

Reply:
Added to the top-level doc.

Reply:
Code seems fine as is. On corner case issues such as this (not too many uses of `@define` in TS), it's also been possible to just sync and fix users in one CL, which I think would save us the intermediary state (and the `enum` and `Map` here).

Reply:
I think the enum and map have value on their own.  I'm maybe a little concerned that if we mark it forbidden, then the `@define` is completely gone: `/** @define {string} */` would end up as `/** @type {string} */`, which means something very different.  But maybe it's fine because it should lead to an error somewhere later, I guess.

Reply:
And it's just a warning here, rather than an error.

I notice that this also causes things like `/** @const {string} */` to lose their constness, etc.  Would we be open to retaining the annotation as if it had had no type (so ignoring the braces but not the annotation itself) or doing the same sort of checking as I had in an earlier revision (where it's an error if the types don't match)?  This would be tough for `@param` but all the other forbidden tags-with-types would work fine for that check.

Reply:
Within Google (and bazel builds), these warnings are actually hard errors.

Not sure what you mean with `@const` or similar tags being dropped, they are currently retained in the output, aren't they?

The general idea here is if at all possible avoid a discrepancy between types seen by Closure and types on the TypeScript level. For comments, that means if at all possible, scrubbing or error'ing on all Closure annotations that contain types.

Ideally I think we'd remove all load-bearing comments on the TS level. It's confusing to have first level syntax for types, but still needing to fall back to comments, plus comments are very fickle. That'll require design time decorators or similar (for things such as `@export`). That's a longer term project though.

In either case, I'm happy to land this code as is, but I'm not clear on the value of the `enum`/`Map` indirection. It makes sense if we anticipate adding more tags with special behavior in the long run (not just `suppress`); do you have a plan there? Otherwise I think the plain `if` statement captures better what's going on. I'm happy to make that change myself though.